rfcs/0167-maintainer-account.md

```diff
@@ -0,0 +1,270 @@
+It is quite simple for users to add themselves to the [maintainer list](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix), and they are expected to do so in a commit prior to adding themselves to a package or module.
+
+The existing [maintainers/README.md](https://github.com/NixOS/nixpkgs/tree/master/maintainers/README.md) in nixpkgs, and the associated maintainer list, spell out a few expectations and requirements for maintainers.
+The requirements in these documents may not be exhaustive and they currently allow for maintainers to opt out of one or more pieces of contact information.
+For example, a maintainer may choose not to provide their user information for the primary nixpkgs hosting platform.
+This opting out for one or more individuals can put extra maintenance overhead on the remaining maintainers; it fragments information which may be important for the overall community; and may even hide information from others working on nixpkgs.
+
+# Detailed design
+
+[design]: #detailed-design
+
+At the time of this writing, the primary nixpkgs hosting platform is GitHub, but this RFC is meant to be platform agnostic.
+If in the future nixpkgs moves to an alternative platform, these requirements and expectations should transfer to the new platform as well.
+
+## Update maintainers/README
+
+Update the section `How to lose maintainer status` in the [maintainers/README.md](https://github.com/NixOS/nixpkgs/tree/master/maintainers/README.md)` file:
+
+```md
+### How to become a maintainer
+
+We encourage people who care about a package to assign themselves as a maintainer.
+Commit access to the nixpkgs repository is not required to become a maintainer.
+
+Maintainers are required to have an account on the nixpkgs hosting platform, which is currently GitHub.
+You are expected to maintain an active account on the nixpkgs hosting platform, and are required to provide this information when adding yourself to `maintainer-list.nix`.
+
+In order to do become a maintainer, add yourself to the [`maintainer-list.nix`](./maintainer-list.nix), and then to the desired package's `meta.maintainers` list, and send a PR with the changes.
+
+### How to lose maintainer status
+
+The following spells out reasons why a maintainer may lose this status.
+This removal ensures an accurate view of the state of maintenance of packages, including who will update the package or respond to issues and pull requests.
+If a maintainer is removed for inactivity or unreachability, this is not permanent and can be reverted by the maintainer at any time.
+
+#### Inactivity
+
+Maintainers who have become inactive on a given package will be removed as maintainer of the package in question.
+
+If we notice that the author hasn't reacted to package-related notifications for more than 3 months, or 2 package releases, whichever is longer, then they can be removed.
```

Comment:
This is very difficult to measure, as it is. Also what happens in the case of multiple maintainers? Does everyone have to reply to every notification in order to game the system?

Reply:
Moreover, what about vacations, some maintainers can simply get burnt out for a while, and some packages release incredibly quickly as well, so what about those packages?

Reply:
While I modified the language here slightly, I'd like to note that some of this is already in the existing README. https://github.com/NixOS/nixpkgs/tree/f0701b576aeea14c09352345cabad4a0b6652c59/maintainers#how-to-lose-maintainer-status

I will change this RFC to be a diff of this document as this isn't the first time this confusion has arisen.

Reply:
> this is already in the existing README

… which kind of illustrates that an RFC doesn't seem to be needed.